Source/Core/VideoBackends/D3D/VertexManager.cpp

```diff
@@ -194,6 +201,9 @@ void VertexManager::ResetBuffer(u32 stride)
 {
 	s_pCurBufferPointer = s_pBaseBufferPointer;
 	IndexGenerator::Start(GetIndexBuffer());
+
+	if (bpmem.genMode.zfreeze)
+		PixelShaderManager::SetZSlope(ZSlope.dfdx, ZSlope.dfdy, ZSlope.f0);
```

Comment:
Why is this in reset? Imo it's fine to move this to line 186 directly into an else block. The same for OGL.